uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -52,10 +52,14 @@ define([
           public_holiday: isPublicHoliday || false
         };
 
-        deferred.resolve(this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params, false)
+        this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params)
           .then(function (data) {
-            return data.values;
-          }));
+            if(data && data.is_error == 1){
```

Comment:
@dhavalsharma I have updated the below comment with proper example, the idea is to send just the error message instead of the whole object. So in this case as we dont have an error_message sent from the server, we can specify a custom error message.

Example:
`deferred.reject(<custom error message>);`

For reference
https://github.com/civicrm/civihr/pull/1327/files#diff-2580b71d3d2a79bd0a3154318a126cb7R74